core/src/main/java/com/graphhopper/reader/osm/OSMTurnRestriction.java

```diff
@@ -24,40 +24,50 @@
  *
  * @author Karl Hübner
  */
-public class OSMTurnRelation {
+public class OSMTurnRestriction {
     private final long fromOsmWayId;
-    private final long viaOsmNodeId;
+    private final ArrayList<Long> viaOSMIds;
     private final long toOsmWayId;
-    private final Type restriction;
+    private final RestrictionType restriction;
+    private final ViaType viaType;
     // vehicleTypeRestricted contains the dedicated vehicle type
     // example: restriction:bus = no_left_turn => vehicleTypeRestricted = "bus";
     private String vehicleTypeRestricted;
     private List<String> vehicleTypesExcept;
 
-    public OSMTurnRelation(long fromWayID, long viaNodeID, long toWayID, Type restrictionType) {
+    public OSMTurnRestriction(long fromWayID, ArrayList<Long> viaOSMIds, long toWayID, RestrictionType restrictionType, ViaType viaType) {
         this.fromOsmWayId = fromWayID;
-        this.viaOsmNodeId = viaNodeID;
+        this.viaOSMIds = viaOSMIds;
         this.toOsmWayId = toWayID;
         this.restriction = restrictionType;
+        this.viaType = viaType;
         this.vehicleTypeRestricted = "";
         this.vehicleTypesExcept = new ArrayList<>();
     }
+    
+    public OSMTurnRestriction(long fromWayID, long viaId, long toWayID, RestrictionType restrictionType, ViaType viaType) {
+    	this(fromWayID, new ArrayList<>(Arrays.asList(viaId)), toWayID, restrictionType, viaType);
+    }
 
     public long getOsmIdFrom() {
         return fromOsmWayId;
     }
-
+    
     public long getOsmIdTo() {
         return toOsmWayId;
     }
 
-    public long getViaOsmNodeId() {
-        return viaOsmNodeId;
-    }
-
-    public Type getRestriction() {
+	public ArrayList<Long> getViaOSMIds() {
+		return viaOSMIds;
+	}
+    
+    public RestrictionType getRestriction() {
         return restriction;
     }
+    
+	public ViaType getViaType() {
+		return viaType;
+	}
```

Comment:
When I use your code formatting xml in eclipse, it somehow changes lots of other pieces of code which I don't manipulate. Thats why I think it might be better if one of you does the formatting once after the functionality is implemented.

Reply:
I'll check if I'm allowed to install community IntelliJ for at least the formatting

Reply:
IANAL but Jetbrains is pretty clear about its usage terms: [Can I use Community Editions of JetBrains IDEs for developing commercial proprietary software?](https://sales.jetbrains.com/hc/en-gb/articles/360021922640-Can-I-use-Community-Editions-of-JetBrains-IDEs-for-developing-commercial-proprietary-software-)

tl;dr you should be safe even if you use it in a business environment

Reply:
I formatted with IntelliJ, it should now be fine